agent/agent.go

```diff
@@ -449,19 +449,25 @@ func (r *agent) runChecks(ctx context.Context) *pb.NodeStatus {
 // GetTimeline returns the current cluster timeline.
 func (r *agent) GetTimeline(ctx context.Context, params map[string]string) ([]*pb.TimelineEvent, error) {
 	if hasRoleMaster(r.Tags) {
-		return r.Timeline.GetEvents(ctx, params)
+		return r.ClusterTimeline.GetEvents(ctx, params)
 	}
 	return nil, trace.BadParameter("requesting cluster timeline from non master")
 }
 
-// RecordTimeline records the events into the cluster timeline.
-func (r *agent) RecordTimeline(ctx context.Context, events []*pb.TimelineEvent) error {
+// RecordClusterEvents records the events into the cluster timeline.
+// Cluster timeline can only be updated if agent has role 'master'.
```

Comment:
This might be a good idea since Agent is starting to look a little bloated. Not sure how big a change this will require. Will probably make changes in a separate PR.